joomla/joomla-cms
# [4.0] [RFC] SVG icons

Pull Request for Issue #28262 .

### Summary of Changes
Introduce a new API for inline svg icons.
Usage:
```html
<?php
echo Factory::getDocument()->setIcon(
	[
		'provider'   => 'fontawesome-free',
		'group'      => 'solid',
		'icon'       => 'user',
		'classes'    => 'demo-class',
		'text'       => Text::_('MOD_LOGIN_VALUE_USERNAME'),
		'attributes' => [
			'data-foo' => 'bar',
			'id'       => 'foobar'
		]
	]
);
?>
```

### Why?
Just google svg vs font icons (here is the first result: https://www.keycdn.com/blog/icon-fonts-vs-svgs , but there are more articles all pointing out that svg's are the preferred way nowadays)

### Why not an implementation with Layouts?
Read my comment here: https://github.com/joomla/joomla-cms/pull/28262#issuecomment-596202930
Also to add one more thing this API is just extending the already existing API for images: `HTMLHelper::image()`

### Testing Instructions
Apply the Patch, run `npm ci`
Check the landing page on the Cassiopeia.
You should get this: 
![Screenshot 2020-03-15 at 15 39 09](https://user-images.githubusercontent.com/3889375/76703627-2a2dd200-66d3-11ea-947b-4b080d2097a4.png)


### Documentation Changes Required
Yes the Api needs documentation, eg usage cases and also 3rd party issuing svg packs


@ciar4n @wilsonge @laoneo
